torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2196,6 +2196,29 @@ at::Tensor& XLANativeFunctions::mv_out(const at::Tensor& self,
   return out;
 }
 
+at::Tensor XLANativeFunctions::nan_to_num(const at::Tensor& self,
+                                          c10::optional<double> nan,
+                                          c10::optional<double> posinf,
+                                          c10::optional<double> neginf) {
+  XLA_FN_COUNTER("xla::");
+  auto element_type = TensorTypeToRawXlaType(self.scalar_type());
```

Comment:
We should do the floating point check here instead. If the `self` is not a floating type, I think you can return a copytensor(self)](https://github.com/pytorch/xla/blob/08ae1044c2a7e314895f9946104cbe399e096515/torch_xla/csrc/torch_util.h#L10) since no update is required.